### PR TITLE
Feat/custom methods

### PR DESCRIPTION
**Pull request by adebsa2401**
In this PR,
- I added a method that returns the 3 most recent posts for a given user in `User` model
- I added a method that updates the posts counter for a user in `Post` model
- I added a method which returns the 5 most recent comments for a given post in `Post` model
- I added a method that updates the comments counter for a post in `Comment` model
- I added a method that updates the likes counter for a post in `Like` model
Cheers

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+
```
Comment (franclobo) [inline app/models/post.rb]: - [ ] A post also has many likes. Please 🙏🏽 add the association declaration between the Posts and Likes to set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (franclobo) [inline app/models/post.rb]: - **[OPTIONAL]** We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private 👇🏽 

image

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (franclobo) [inline app/models/like.rb]: - **[OPTIONAL]** We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_likes_counter method every time a new like has been saved to database, then make the update_likes_counter method private 👍🏽

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (franclobo) [inline app/models/comment.rb]: - **[OPTIONAL]** We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new like has been saved to database, then make the update_comments_counter method private 👍🏽

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @adebsa2401, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors ✔️ 
- Good models 💯 
- Correct methods ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please 🙏🏽 fix the migrations to be able to use your app 👇🏽 

image

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED 🥇 

Hi @adebsa2401 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Use of Rails Active Callback `after_save` 👍 
:heavy_check_mark: Associatioans setup correctly

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions

 **[ OPTIONAL]**
- Kindly remove the text below the MIT license section in your ReadMe. That text was documentation to help you add the correct MIT license link. Now that you that added correctly, you can remove the text
Screen Shot 2023-02-20 at 7 40 29 PM
 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**